solver/src/metrics.rs

```diff
@@ -0,0 +1,107 @@
+use std::{convert::TryInto, time::Instant};
+
+use anyhow::Result;
+use model::order::Order;
+use prometheus::{IntCounter, IntCounterVec, IntGaugeVec, Opts, Registry};
+use strum::{AsStaticRef, VariantNames};
+
+use crate::liquidity::Liquidity;
+
+pub trait SolverMetrics {
+    fn liquidity_fetched(&self, liquidity: &[Liquidity]);
+    fn settlement_computed(&self, solver_type: &str, start: Instant);
+    fn order_settled(&self, order: &Order);
+}
+
+// TODO add labeled interaction counter once we support more than one interaction
+pub struct Metrics {
+    trade_counter: IntCounter,
+    order_settlement_time: IntCounter,
+    solver_computation_time: IntCounterVec,
+    liquidity: IntGaugeVec,
+}
+
+impl Metrics {
+    pub fn new(registry: &Registry) -> Result<Self> {
+        let trade_counter =
+            IntCounter::new("gp_v2_solver_trade_counter", "Number of trades settled")?;
+        registry.register(Box::new(trade_counter.clone()))?;
+
+        let order_settlement_time = IntCounter::new(
+            "gp_v2_solver_order_settlement_time_seconds",
+            "Counter for the number of seconds between creation and settlement of an order",
+        )?;
+        registry.register(Box::new(order_settlement_time.clone()))?;
+
+        let solver_computation_time = IntCounterVec::new(
+            Opts::new(
+                "gp_v2_solver_computation_time_ms",
+                "Ms each solver takes to compute their solution",
+            ),
+            &["solver_type"],
+        )?;
```

Comment:
Also seems like it should be a histogram unless I can't imagine the right visualization. I don't mind changing these types later either if we want to merge this and make a dashboard first.

Reply:
Here is was afraid that different solvers might have very different values so finding good bucket sizes is even harder than for the order settlement time.